asf_levies_model/notebooks/report_scenarios_summary_v2.py

```diff
@@ -0,0 +1,755 @@
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.2
+#   kernelspec:
+#     display_name: asf_levies_model
+#     language: python
+#     name: python3
+# ---
+
+# %%
+import pandas as pd
+import copy
+from datetime import datetime
+from asf_levies_model.getters.load_data import (
+    download_annex_4,
+    download_annex_9,
+    process_data_RO,
+    process_data_AAHEDC,
+    process_data_GGL,
+    process_data_WHD,
+    process_data_ECO,
+    process_data_FIT,
+    process_tariff_elec_other_payment_nil,
+    process_tariff_elec_other_payment_typical,
+    process_tariff_gas_other_payment_nil,
+    process_tariff_gas_other_payment_typical,
+    ofgem_archetypes_data,
+    ofgem_archetypes_scheme_eligibility,
+    ofgem_archetypes_net_income_deciles_full,
+)
+
+from asf_levies_model.levies import RO, AAHEDC, GGL, WHD, ECO, FIT
+
+from asf_levies_model.tariffs import ElectricityOtherPayment, GasOtherPayment
+
+from asf_levies_model.summary import (
+    _rebalance_levies,
+    create_scenario_weights_dict,
+    calculate_cost_stream,
+    set_common_denominators,
+    calculate_fuel_poverty_rates,
+)
+
+from asf_levies_model import config, PROJECT_DIR
+
+from asf_levies_model.consumers import Consumer
+
+# %% [markdown]
+# **Status quo levies from Annex 4**
+
+# %%
+# Assign denominator values
+supply_elec = 94_200_366
+supply_gas = 265_197_947
+customers_gas = 24_503_683
+customers_elec = 29_078_770
+
+# %%
+# Scaling factor for estimating domestic share of FIT revenue
+total_supply_elec = (
+    250_020_739  # DESNZ GB total electricity consumption - all meters (2022)
+)
+exempt_eii_supply = 9_417_916  # Oct-Dec2024 period, Annex 4, New FIT methodology tab
+fit_scaling_factor = supply_elec / (total_supply_elec - exempt_eii_supply)
+
+# %%
+# Instantiate status quo levies with Annex 4 data
+fileobject = download_annex_4(as_fileobject=True)
+levies = [
+    RO.from_dataframe(process_data_RO(fileobject), denominator=supply_elec),
+    AAHEDC.from_dataframe(process_data_AAHEDC(fileobject), denominator=supply_elec),
+    GGL.from_dataframe(process_data_GGL(fileobject), denominator=customers_gas),
+    WHD.from_dataframe(process_data_WHD(fileobject)),
+    ECO.from_dataframe(process_data_ECO(fileobject)),
+    FIT.from_dataframe(
+        process_data_FIT(fileobject),
+        scaling_factor=fit_scaling_factor,
+    ),
+]
+fileobject.close()
+
+# %%
+# Create dictionary of denominators for each levy
+denominators = set_common_denominators(
+    levies, supply_elec, supply_gas, customers_gas, customers_elec
+)
+
+# %% [markdown]
+# **Defining rebalancing weights for each scenario**
+
+# %%
+# Scenario 1: Status quo weights which reflect denominators
+status_quo = create_scenario_weights_dict(levies)
+
+# Manually update WHD weights according to denominator balance
+status_quo["whd"]["new_electricity_weight"] = denominators["whd"]["customers_elec"] / (
```

Comment:
I've operationalised adding the gas and electricity customer numbers to the WHD constructor, so you don't necessarily have to do this to create the weights.

In future, when loading WHD you can do:
`WHD.from_dataframe(process_data_WHD(fileobject), customers_gas=customers_gas, customers_elec=customers_elec)`